### PR TITLE
STEP-91: remove Google Form, show all eligible countries in all locales, and add a country selector when entering a phone number

### PR DESCRIPTION
**Pull request by jay-hodgson**
and some extra spaces in the dutch translation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,11 @@
   },
   "dependencies": {
     "babel-polyfill": "^6.26.0",
-    "vue": "^2.5.16",
+    "vue": "2.6.10",
```
Comment (jay-hodgson) [inline package.json:16]: build conflict, dependency build with 2.6.10 where we were building with 2.6.12?

File: src/components/study/Eligibility.vue [hunk 2]
```diff
@@ -92,10 +92,6 @@ export default {
       this.step += 1;
     },
     doSubmit() {
-      if (this.comfort === this.$t('eligibility-screen.q3.have-android') && window.queryParams.android !== "true") {
```
Comment (jay-hodgson) [inline src/components/study/Eligibility.vue:95]: removal of the Google Form

File: src/components/study/Registration.vue [hunk 3]
```diff
@@ -25,27 +29,30 @@
...
 export default {
   name: 'StudyRegistration',
-  components: { MainNav },
+  components: { MainNav, VueTelInput },
   data() {
     return {
-      phone: '',
-      showAndroid: window.queryParams.android === "true"
+      phoneValue: '',
+      countryCodes: [this.$t('registration-screen.other-region-code'), 'US'],
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue]: only show 2 countries, US and another one (Canada or Netherlands)

File: src/components/study/Registration.vue [hunk 4]
```diff
@@ -25,27 +29,30 @@
...
+      defaultCountry: this.$i18n.locale == 'en' ? 'US' : this.$t('registration-screen.other-region-code'),
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue]: Default to US if in en, otherwise the other country.

File: src/components/study/Registration.vue [hunk 5]
```diff
@@ -60,12 +67,11 @@ export default {
...
       var snackbar = this.$refs.snackbar
-      var phoneFormatted = this.phone.replace(/[^\d]/g,'')
-
-      this.$store.setPhone(this.phone)
+      var phoneFormatted = this.phoneValue.replace(/[^\d]/g,'')
+      this.$store.setPhone(this.phoneValue)
       axios.post('https://webservices.sagebridge.org/v3/itp', {
         studyId: 'sage-mpower-2',
-        phone: {number: phoneFormatted, regionCode: 'US'},
+        phone: {number: phoneFormatted, regionCode: this.selectedRegionCode},
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue:73]: pass the selected region code to this call

File: src/components/study/Registration.vue [hunk 6]
```diff
@@ -2,17 +2,21 @@
...
     <div class="container">
-      <p>Thank you! Please enter your mobile phone number <!--and select your phone type--> to receive your download link for the mPower app:</p>
+      <p>Thank you! Please enter your mobile phone number and select your phone type to receive your download link for the mPower app:</p>
 
-      <p style="text-align: center; margin-top: 2rem">
-        <mdc-textfield ref="phoneField" v-model="phone" label="Enter phone number" type="tel" pattern="[0-9]*"/>
-      </p>
-      <p v-if="showAndroid" style="text-align: center; margin-top: 3rem">Pick one depending on your mobile phone:</p>
+      <div style="text-align: center; margin-top: 2rem">
+        <vue-tel-input
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue:8]: Use vue-tel-input, capture the user provided phone number in phoneValue, set the default country based on the locale, and restrict the available countries to the 2 we say are eligible in this study in this locale (US and Canada, or US and Netherlands)

Comment (jay-hodgson) [reply src/components/study/Registration.vue:8]: When the country is changed, call onCountryChange()

File: src/components/study/Registration.vue [hunk 7]
```diff
@@ -2,17 +2,21 @@
...
+          :onlyCountries="countryCodes"
+          @country-changed="onCountryChange"></vue-tel-input>
+      </div>
+      <p style="text-align: center; margin-top: 3rem">Pick one depending on your mobile phone:</p>
       <div class="buttons">
         <a @click="apple">
           <BridgeImage src="/static/images/App_Store_Badge.svg" :style="{opacity: hasNumber}"/>
         </a>
-        <a @click="google" v-if="showAndroid">
+        <a @click="google">
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue:19]: Always show

File: src/components/study/Registration.vue [hunk 8]
```diff
@@ -25,27 +29,30 @@
...
   computed: {
     hasNumber: function() {
-      return this.phone.replace(/\D/g,'').length >= 10 ? 1 : .5
-    }
+      return ( this.phoneValue && this.phoneValue.replace(/\D/g,'').length >= 10 ) ? 1 : .5
+    },
   },
   methods: {
+    onCountryChange: function (countryObject) {
+      this.selectedRegionCode = countryObject.iso2
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue:52]: Update the local selectedRegionCode with this value (this works, but is this is the right way of doing this in this framework when the external component provides an event handle hook?)

File: src/components/study/Eligibility.vue [hunk 9]
```diff
@@ -92,11 +92,7 @@ export default {
...
-          window.location = ANDROID_FORM_LINK;
-          return;
-      }
-      var validResidence = (this.residence === this.$t('eligibility-screen.q2.in-US') || this.residence === this.$t('eligibility-screen.q2.in-other'));
+      var validResidence = (this.residence !== this.$t('eligibility-screen.q2.outside-eligible-area'));
```
Comment (jay-hodgson) [inline src/components/study/Eligibility.vue:95]: all other options are eligible, so verify residence is not the ineligible choice

File: src/components/study/Registration.vue [hunk 10]
```diff
@@ -25,27 +29,29 @@
...
   name: 'StudyRegistration',
-  components: { MainNav },
+  components: { MainNav, VueTelInput },
   data() {
     return {
-      phone: '',
-      showAndroid: window.queryParams.android === "true"
+      phoneValue: '',
+      countryCodes: ['CA','US','NL'],
+      defaultCountry: this.$t('registration-screen.default-region-code'),
```
Comment (jay-hodgson) [inline src/components/study/Registration.vue:41]: Make Canada, US, and Netherlands the only available options.  Set the default region code based on the locale (in nl it's set to NL, in en it's US).

File: src/components/study/Registration.vue (hunk 8 above)
Comment (alxdarksage) [reply src/components/study/Registration.vue:52]: I think so, yes. There is an activeCountryCode property in the component but I'm not sure how you'd bind to it, possibly :selectedRegionCode="activeCountryCode"

**Review by alxdarksage — APPROVED** [review]

**Pull request merged**